console/console-server/src/main/resources/iot.schema.graphql

```diff
@@ -0,0 +1,231 @@
+  Command: AddressConfig_iot_enmasse_io_v1alpha1!
+}
+
+
+type managedDownstreamStrategy_iot_enmasse_io_v1alpha1 {
+  addressSpace: AddressSpaceConfig_iot_enmasse_io_v1alpha1!
+  addresses: AddressesConfig_iot_enmasse_io_v1alpha1!
+}
+
+type externalDownstreamStrategy_iot_enmasse_io_v1alpha1 {
+  connectionInformation: connectionInformation_iot_enmasse_io_v1alpha1!
+}
+
+type IotProjectSpec_iot_enmasse_io_v1alpha1 {
+  downstreamStrategy: iotProjectDownStreamStrategy !
+  configuration: String!
+}
+
+type IoTProject_iot_enmasse_io_v1alpha1 {
+
+  metadata: ObjectMeta_v1! @goField(name: "ObjectMeta")
+  enabled: Boolean!
+  spec: IotProjectSpec_iot_enmasse_io_v1alpha1!
+  status: IoTProjectStatus_iot_enmasse_io_v1alpha1!
+  devices(iotproject: String!, first: Int, offset: Int, filter: String, orderBy: String): devicesQueryResult_consoleapi_iot_enmasse_io_v1alpha1
+  adapters: [adapter]
+}
+
+interface adapter {}
+
+# TODO : map to k8s objectcs and pull endpoints details from openshift routes ?
+type HTTP_Adapter implements adapter {
+  url : String!
+}
+
+type MQTT_Adapter implements adapter {
+ tls_enabled: Boolean!
+ host : String!
+ port : Int!
+}
+
+type AMQP_Adapter implements adapter {
+  url : String!
+}
+
+type CoAP_Adapter implements adapter {
+  url : String!
+}
+
+type device {
+  deviceID: String!
+  enabled: Boolean!
+  viaGateway: Boolean!
+  jsonData: String! #The Json representation of this device.
+}
+
+type credential{
+  authID: String!
+  enabled: Boolean!
+  type: iotCredentials!
+  ext : String #A Json Object.
+  secrets: String! #A Json Array containing the secrets.
+}
+
+type devicesQueryResult_consoleapi_iot_enmasse_io_v1alpha1 {
+  total: Int!
+  devices: [device!]!
+}
+
+type credentialsQueryResult_consoleapi_iot_enmasse_io_v1alpha1 {
+  total: Int!
+  credentials: [credential!]!
+}
+
+type addressSpaceAndIotProjectsQueryResult_consoleapi_iot_enmasse_io_v1alpha1 {
+  total: Int!
+  addressSpaces: [AddressSpace_consoleapi_enmasse_io_v1beta1!]
+  iotProjects: [IoTProject_iot_enmasse_io_v1alpha1!]
+}
+
+enum projectQueryType {
+  iotProject
+  addressSpace
+}
+
+type Query {
+
+  "Returns the namespaces and iotprojects visible to this user, optionaly filtered by project type"
+  #This extends the existing "addressSpaces" query
+  allProjects(first: Int, offset: Int, filter: String, orderBy: String, projectType: projectQueryType): addressSpaceAndIotProjectsQueryResult_consoleapi_iot_enmasse_io_v1alpha1!
+
+  "Returns the iot project tied to the given nameSpace"
+  iotProject(namespace: String!): IoTProject_iot_enmasse_io_v1alpha1!
+
+  device(iotproject: String!, deviceId: String!): device!
+  devices(iotproject: String!, first: Int, offset: Int, filter: String, orderBy: String): devicesQueryResult_consoleapi_iot_enmasse_io_v1alpha1
+  credentials(iotproject: String!, deviceId: String!): credentialsQueryResult_consoleapi_iot_enmasse_io_v1alpha1!
+}
+
+#
+# Inputs Types
+#
+
+
+enum deviceRegistryMutationResult{
```

Comment:
Have you seen http://spec.graphql.org/draft/#sec-Errors.Error-result-format?

Reply:
No, I missed it. Thanks !